Issue 210: Require hidden explanatory text for macros that rely on data

SebastianZ opened the pull request:
The added test checks whether there is an explanation for each macro (by checking whether the sibling element before the element containing the macro has `class="hidden"`) and whether that explanation has a link (i.e. an `<a href="http...">` element; it doesn't check whether the link is valid).

Sebastian

Elchi3 commented on locale/de.properties:
```diff
@@ -106,6 +106,12 @@ http_links=HTTP Links
+data_macro_explanation=Datenmakro Erklärung
+
+# Description of test checking the explanation of data macros linking to the data source
+data_macro_explanation_desc=Datenmakros benötigen eine Erklärung über ihnen, die einen Link zur Datenquelle enthält.
```
Besser: "Oberhalb von Datenmakros muss ein versteckter Hinweistext stehen, der einen Link zur Datenquelle enthält."

Elchi3 commented on locale/de.properties:
```diff
@@ -175,6 +181,12 @@ article_length_desc=Hinweise zur Länge des Artikels
 link_using_wrong_locale=Link %s verwendet falsche Übersetzung. '%s' erwartet.
 
+# Error message for data macro missing an explanation above it linking to the data source
+data_macro_explanation_missing=Dem Datenmakro %s fehlt eine Erklärung, die zur Datenquelle verweist. Die Erklärung muss durch 'class="hidden"' versteckt sein.
```
statt Erklärung: "ein Hinweistext, der" und "Der Hinweis muss ..."

Elchi3 commented on locale/de.properties:
```diff
@@ -175,6 +181,12 @@ article_length_desc=Hinweise zur Länge des Artikels
+data_macro_explanation_missing=Dem Datenmakro %s fehlt eine Erklärung, die zur Datenquelle verweist. Die Erklärung muss durch 'class="hidden"' versteckt sein.
+
+# Error message for data macro explanation missing a link referring to the data source
+data_macro_source_link_missing=Der Erklärung für das Datenmakro %s fehlt ein Link zur Datenquelle.
```
Dem Hinweistext

Elchi3 commented on data/tests/data-macro-explanation.js:
```diff
@@ -0,0 +1,46 @@
+const dataMacros = /^(?:compat|css_ref|cssanimatedproperties|cssinfo|csssyntax|webextbrowsercompat)$/i;
```
Please add a specification text to the top of the file:

<pre>
/*
 *  Title: Test that data macros have a note above them pointing to data sources.
 *
 *  Example 1: If you are using the e.g. the {{compat}} or {{cssinfo}} macros, there should 
 *  be a hidden note for contributors in edit-mode, so that they know where the data comes 
 *  from and how to change it. 
 * 
 *  Implementation notes: This test checks if there is a note for each macro 
 *  (by checking whether the sibling element before the element containing the macro
 *  has class="hidden") and whether that note contains a link (i.e. an &lt;a href="http..."&gt; 
 *  element; it doesn't check whether the link is valid).
 * 
 */
</pre>

Elchi3 replied:
In the implementation notes maybe also say that each data macro usage requires a hidden note as opposed to one note for several data macro uses.

SebastianZ replied:
"if there is a note for each macro" sounds clear enough, IMO.

SebastianZ commented:
Thank you for the quick review! I've addressed your comments in https://github.com/Elchi3/mdn-doc-tests/pull/212/commits/ee2d8a733520a71f845ce66f1c8c2ee075567a39.

Sebastian
